cvat/apps/engine/cloud_provider.py

```diff
@@ -31,10 +32,27 @@
 from cvat.apps.engine.models import CloudProviderChoice, CredentialsTypeChoice
 from cvat.apps.engine.utils import get_cpu_number
 
+class NamedBytesIO(BytesIO):
+    @property
+    def filename(self) -> Optional[str]:
+        return getattr(self, '_filename', None)
+
+    @filename.setter
+    def filename(self, value: str) -> None:
+        self._filename = value
+
 slogger = ServerLogManager(__name__)
 
 ImageFile.LOAD_TRUNCATED_IMAGES = True
 
+CPU_NUMBER = get_cpu_number()
+
+def normalize_threads_number(threads_number: Optional[int], number_of_files: int) -> int:
+    if threads_number is None:
+        return min(CPU_NUMBER, settings.CVAT_MAX_THREADS_NUMBER_FOR_DATA_DOWNLOADING, max(math.ceil(number_of_files / settings.CVAT_NUMBER_OF_FILES_PER_THREAD), 1))
+
+    return min(threads_number, CPU_NUMBER, settings.CVAT_MAX_THREADS_NUMBER_FOR_DATA_DOWNLOADING)
+
```

Comment:
@coderabbitai, they are configurable now since these settings were moved into settings/base.py. I don't think that it is necessary to add such env variables. Please check the settings one more time and don't repeat other people's comments.